src/Core/NotificationCenter/Commands/MarkNotificationDeletedCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿#nullable enable
+using Bit.Core.Context;
+using Bit.Core.Exceptions;
+using Bit.Core.NotificationCenter.Authorization;
+using Bit.Core.NotificationCenter.Commands.Interfaces;
+using Bit.Core.NotificationCenter.Entities;
+using Bit.Core.NotificationCenter.Repositories;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.NotificationCenter.Commands;
+
+public class MarkNotificationDeletedCommand : IMarkNotificationDeletedCommand
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly IAuthorizationService _authorizationService;
+    private readonly INotificationRepository _notificationRepository;
+    private readonly INotificationStatusRepository _notificationStatusRepository;
+
+    public MarkNotificationDeletedCommand(ICurrentContext currentContext,
+        IAuthorizationService authorizationService,
+        INotificationRepository notificationRepository,
+        INotificationStatusRepository notificationStatusRepository)
+    {
+        _currentContext = currentContext;
+        _authorizationService = authorizationService;
+        _notificationRepository = notificationRepository;
+        _notificationStatusRepository = notificationStatusRepository;
+    }
+
+    public async Task MarkDeletedAsync(Guid notificationId)
+    {
+        if (!_currentContext.UserId.HasValue)
+        {
+            throw new NotFoundException();
+        }
+
+        var notification = await _notificationRepository.GetByIdAsync(notificationId);
+        if (notification == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var authorizationResult = await _authorizationService.AuthorizeAsync(_currentContext.HttpContext.User,
+            notification, NotificationOperations.Read);
+        if (!authorizationResult.Succeeded)
+        {
+            throw new NotFoundException();
+        }
```

Comment:
:art:
We are using this pattern for most of the new commands. 

Should we leverage the `AuthorizeOrThrowAsync` utility to DRY it up?

https://github.com/bitwarden/server/blob/c94a084c86f9e2a653e3e1562906a2ca8db321c9/src/Core/Utilities/AuthorizationServiceExtensions.cs#L38

Reply:
Fixed in afa22d17